### PR TITLE
Allow running mount_handler tests as part of go test

### PR DESCRIPTION
**Pull request by didrocks**
Hook up running rust tests from go test command, with proper detection and cache handling.

### CODE REVIEW

File: internal/policies/mount/adsys_mount/run_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	}
+}
+
+func TestMain(m *testing.M) {
+	canRun, withCoverage := testutils.CanRunRustTests()
+	if !canRun {
+		os.Exit(1)
+	}
+
+	_ = withCoverage
```
Comment (denisonbarbosa) [inline internal/policies/mount/adsys_mount/run_test.go:33]: I know this is being set up for later, but can't we just leave at ```canRun, _ := testutils.CanRunRustTests()``` plus a comment informing that coverage will be "coming soon"?

File: internal/policies/mount/adsys_mount/run_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	// properly inform Go about which files to use for cache by reading input files.
+	testutils.MarkRustFilesForTestCache(t)
+
+	cmd := exec.Command("cargo", "test")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		t.Fail()
```
Comment (denisonbarbosa) [inline internal/policies/mount/adsys_mount/run_test.go:23]: Shouldn't we log the error here as well (even though we are already setting the cmd stds to the os ones)?

**Review by denisonbarbosa — CHANGES_REQUESTED** [review]
Looks good to me, just a few questions about some stuff

File: internal/policies/mount/adsys_mount/run_test.go (hunk 2 above)
Comment (didrocks) [reply internal/policies/mount/adsys_mount/run_test.go:23]: No really, I tried it with failing tests and then, the error message makes sense: you see that this test fails, and you get the old stdout/stderr printed above, which is what makes sense and is in line with pure Go tests.

**Comment by didrocks** [comment]
Merged your suggestion and answered your comment

**Review by denisonbarbosa — APPROVED** [review]
LGTM! Feel free to merge.

**Pull request merged**